### PR TITLE
File fetching and url genering

### PR DESCRIPTION
**Pull request by bbrands02**

### CODE REVIEW

File: src/ActionHandler/ViewFileHandler.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    /**
+     *  This function returns the requered configuration as a [json-schema](https://json-schema.org/) array.
+     *
+     * @throws array a [json-schema](https://json-schema.org/) that this  action should comply to
+     */
+    public function getConfiguration(): array
+    {
+        return [
+            '$id'         => 'https://commongateway.nl/pdd.SyncCasesAction.action.json',
```
Comment (WilcoLouwerse) [inline src/ActionHandler/ViewFileHandler.php:50]: SyncCasesAction != ViewFileHandler

File: src/Command/SyncXxllncCasesCommand.php [hunk 2]
```diff
@@ -117,9 +117,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             // if ($this->syncXxllncCasesService->getZaak($action->getConfiguration(), $caseId) === true) {
             // return Command::FAILURE;
             // }
-            isset($style) === true && $style->info("Succesfully synced and created a WOO object from xxllnc case: $caseId.");
+            isset($style) === true && $style->error("Single object synchronization not supported yet.");
 
-            return Command::SUCCESS;
+            return Command::FAILURE;
```
Comment (WilcoLouwerse) [inline src/Command/SyncXxllncCasesCommand.php:122]: Does this Command have a succes return somewhere else?

File: src/Service/FileService.php [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+     *
+     * @param Value    $value            The object entity associated with the file.
+     * @param string   $title            File title.
+     * @param string   $base64           Encoded base64.
+     * @param string   $mimeType         Mimetype of file.
+     * @param Endpoint $downloadEndpoint Endpoint to use for downloading the file.
+     *
+     * @return void
+     */
+    public function createOrUpdateFile(Value $value, string $title, string $base64, string $mimeType, Endpoint $downloadEndpoint): string
```
Comment (WilcoLouwerse) [inline src/Service/FileService.php:144]: A lot of properties, could be mashed together in a single array $config property or something like that

File: src/Service/SyncXxllncCasesService.php [hunk 4]
```diff
@@ -169,6 +181,80 @@ private function deleteNonExistingObjects(array $idsSynced, Source $source, stri
...
+     * Handles custom logic for processing and hydrating file fields from the given result.
+     *
+     * @param ObjectEntity $object       The main object entity that will be hydrated.
+     * @param array        $result       The result data that contains the information of file fields.
+     * @param Endpoint     $fileEndpoint The endpoint entity for the file.
+     * @param Source       $source       The source entity that provides the source of the result data.
+     *
+     * @return ObjectEntity              The hydrated object entity.
+     */
+    private function handleCustomLogic(ObjectEntity $object, array $result, Endpoint $fileEndpoint, Source $source)
```
Comment (WilcoLouwerse) [inline src/Service/SyncXxllncCasesService.php:194]: Very big function, could be split in smaller functions

**Review by WilcoLouwerse — APPROVED** [review]

File: src/Command/SyncXxllncCasesCommand.php (hunk 2 above)
Comment (bbrands02) [reply src/Command/SyncXxllncCasesCommand.php:122]: Yes the last return

File: src/Service/FileService.php (hunk 3 above)
Comment (bbrands02) [reply src/Service/FileService.php:144]: Shall be done later

File: src/Service/SyncXxllncCasesService.php (hunk 4 above)
Comment (bbrands02) [reply src/Service/SyncXxllncCasesService.php:194]: Shall be done later

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**